contracts/libraries/Validator.sol

```diff
@@ -58,14 +58,13 @@ library Validator {
             address recoveredConfirmation1 = recover(confirmationHash, slice(confirmSignatures, 65, 65));
 
             return recoveredTx0 == recoveredConfirmation0 && recoveredTx1 == recoveredConfirmation1 &&
-                ((recoveredTx0 != address(0) && recoveredConfirmation0 != address(0)) ||
-                (recoveredTx1 != address(0) && recoveredConfirmation1 != address(0)));
+                (recoveredTx0 != address(0) || recoveredTx1 != address(0));
```

Comment:
I think the "||" should be changed to an "&&".  Otherwise I believe a valid sig/confirmsig of tx0 and a empty sig/confirmSig for tx1 would pass right?

Reply:
@colin-axner @legengliu If you make this change, then the test "Test checkSigs with only second input" fails to pass it's a test that does exactly that: where only one of the confirm sigs are non zero. In order for the change to work, we'd need to remove this test, or check that this test fails.

We might want to add the following checks:
```
if (!input1)
    require(confirmSignatures.length == 65, "one confirm signatures required with one input");
```

Reply:
Addressed